Clinical-Genomics/scout
# Get an alert on caseS page when somebody validates variants you ordered Sanger sequencing for

This PR adds a functionality or fixes a bug.
- Simpler alternative to #4345 to fix #4277

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>


**How to test, locally**:
1. Edit the scout config file by adding the following lines:
 
```
MAIL_USERNAME="clark.kent@mail.com"
MAIL_PASSWORD="password"
ADMINS=["clark.kent@mail.com"]
```
This will allow you to order Sanger sequencing for variants, even if it crashed when you do it, sanger will result ordered anyway

1. Add another user for cust000 -> `scout --demo load user -i cust000 -u "Diana Prince" -m wonderwoman@mail.com`
2. As user clark Kent, order sanger for one variant (top right on variant page)
3. Now if you go to institute page you should see an alert that says you need to validate that variant.
 
<img width="651" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/0828ab66-2be7-423c-9494-b09e61f46946">

4. Log out and log in as the other user
5. As the other user validate the variant either as true positive or false positive
6. Log out and log in again with Clark Kent
7. Go to institute page and make sure you get the alert that the variant was validated and you also need to validate it

<img width="1333" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/9d3ae191-08f7-4a26-9f80-f9f51585d556">

8. Validate the variant as Clark Kent and the alert should disappear

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
